client/src/main/java/io/pravega/client/stream/impl/ReaderGroupImpl.java

```diff
@@ -178,6 +180,11 @@ public void readerOffline(String readerId, Position lastPosition) {
                 .thenApply(checkpoint -> checkpoint); //Added to prevent users from canceling completeCheckpoint
     }
 
+    @Override
+    public CompletableFuture<Checkpoint> initiateCheckpoint(String checkpointName) {
+        return initiateCheckpoint(checkpointName, connectionPool.getInternalExecutor());
```

Comment:
If this is going to be running on our internal thread pool, it would prefer it it was capped so as not to use too much of it in case the user creates a lot of checkpoints. To do this we can use `SequentialProcessor` from common. Just turn the function passed to the delayed task on 199 below into an invocation to add a task to the sequentialProcessor which does the fetchUpdates etc.

Reply:
Done the changes accordingly.